File: exercism/javascript/anagram/anagram.js
const { flow, eq } = require('lodash/fp')


const signature = (word) => {
  const chars = word.toLowerCase().split()
  chars.sort()
  return chars.join('')
}


class Anagram {
  constructor (subject) {
    this.target = signature(subject)
  }
  matches (xs) {
    const match = flow(signature, eq(this.target))
    return xs.filter(match)
  }
}

module.exports = Anagram

File: exercism/haskell/anagram/Anagram.hs
module Anagram (anagramsFor) where

import Data.Char (toLower)
import Data.List (sort)

lower :: String -> String
lower = map toLower

key :: String -> String
key = sort . lower

anagramsFor :: String -> [String] -> [String]
anagramsFor word =
  filter (\ch -> key ch == key word && lower ch /= lower word)
